test/unit/StakingPool/calculatePremium.js

```diff
@@ -0,0 +1,624 @@
+const { expect } = require('chai');
+const { ethers } = require('hardhat');
+const { parseEther } = ethers.utils;
+const { BigNumber } = ethers;
+const {
+  calculateSurgePremiums,
+  calculateBasePrice,
+  divCeil,
+  calculateFixedPricePremium,
+  calculatePriceBump,
+} = require('./helpers');
+const { daysToSeconds } = require('../../../lib/helpers');
+
+describe('calculatePremium', function () {
+  // If an arithmetic operation results in underflow or overflow outside of an unchecked { ... } block.
+  const UNDER_OR_OVERFLOW = 0x11;
```

Comment:
i'd look to see if ethers already has this if not we can also put them in our common utils lib

Reply:
Couldn't find them in ethers. Added them all to `utils/errors.js`